src/cascadia/TerminalSettingsEditor/NullableColorPicker.cpp

```diff
@@ -0,0 +1,233 @@
+        if (!_ColorSchemeVMProperty)
+        {
+            _ColorSchemeVMProperty =
+                DependencyProperty::Register(
+                    L"ColorSchemeVM",
+                    xaml_typename<Editor::ColorSchemeViewModel>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ nullptr });
+        }
+        if (!_CurrentColorProperty)
+        {
+            _CurrentColorProperty =
+                DependencyProperty::Register(
+                    L"CurrentColor",
+                    xaml_typename<Windows::Foundation::IReference<Microsoft::Terminal::Core::Color>>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ nullptr, PropertyChangedCallback{ &NullableColorPicker::_OnCurrentColorValueChanged } });
+        }
+        if (!_ShowNullColorButtonProperty)
+        {
+            _ShowNullColorButtonProperty =
+                DependencyProperty::Register(
+                    L"ShowNullColorButton",
+                    xaml_typename<bool>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ box_value(true) });
+        }
+        if (!_NullColorButtonLabelProperty)
+        {
+            _NullColorButtonLabelProperty =
+                DependencyProperty::Register(
+                    L"NullColorButtonLabel",
+                    xaml_typename<hstring>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ nullptr });
+        }
+        if (!_NullColorPreviewProperty)
+        {
+            _NullColorPreviewProperty =
+                DependencyProperty::Register(
+                    L"NullColorPreview",
+                    xaml_typename<Windows::UI::Color>(),
+                    xaml_typename<Editor::NullableColorPicker>(),
+                    PropertyMetadata{ box_value(Windows::UI::Colors::Transparent()) });
+        }
+    }
+
+    void NullableColorPicker::_OnCurrentColorValueChanged(const DependencyObject& d, const DependencyPropertyChangedEventArgs& /*e*/)
+    {
+        const auto& obj{ d.try_as<Editor::NullableColorPicker>() };
+        get_self<NullableColorPicker>(obj)->_UpdateColorChips();
+    }
+
+    void NullableColorPicker::_UpdateColorChips()
+    {
+        const auto& currentColor = CurrentColor();
+        for (const auto& colorChip : _colorChips)
+        {
+            const auto& chipColor = colorChip.DataContext().as<Editor::ColorTableEntry>().Color();
+            colorChip.IsChecked(currentColor ?
+                                    chipColor.R == currentColor.Value().R && chipColor.G == currentColor.Value().G && chipColor.B == currentColor.Value().B :
+                                    false);
+        }
+    }
+
+    constexpr double _CalculateLuminance(const Windows::UI::Color& color)
+    {
+        auto linearize = [](float value) {
```

Comment:
I'd either change this parameter to `double` or replace all the doubles with floats (= append `f` to all numbers).
This lambda also needs a `static constexpr` since it's static and constant for the duration of the app.